app/models/course/condition/achievement.rb

```diff
@@ -18,6 +18,11 @@ def satisfied_by?(course_user)
     course_user.achievements.exists?(achievement)
   end
 
+  # Array of classes that the condition depends on.
+  def self.dependent_classes
```

Comment:
Should acts_as_condition define a dummy `dependent_classes` which raises NotImplementedError?